diff --git a/build.sh b/build.sh
index 314ed5b..468e91b 100755
--- a/build.sh
+++ b/build.sh
@@ -21,6 +21,7 @@ SUPPORTED_VARIANTS="generic hyperv qemu rootfs virtualbox vmware"
 DEFAULT_ARCH=amd64
 DEFAULT_BRANCH=kali-rolling
 DEFAULT_DESKTOP=xfce
+DEFAULT_HOSTNAME=kali
 DEFAULT_LOCALE=en_US.UTF-8
 DEFAULT_MIRROR=http://http.kali.org/kali
 DEFAULT_TIMEZONE=America/New_York
@@ -32,6 +33,7 @@ ARCH=
 BRANCH=
 DESKTOP=
 FORMAT=
+HOSTNAME=
 KEEP=false
 LOCALE=
 MIRROR=
@@ -195,6 +197,7 @@ create_vm() {
         -t branch:$BRANCH \
         -t desktop:$DESKTOP \
         -t format:$FORMAT \
+        -t hostname:$HOSTNAME \
         -t imagename:$IMAGENAME \
         -t imagesize:$SIZE \
         -t keep:$KEEP \
@@ -236,6 +239,7 @@ Build options:
 Customization options:
   -D DESKTOP  Desktop environment installed in the image, default: $(b $DEFAULT_DESKTOP)
               Supported values: $SUPPORTED_DESKTOPS
+  -H HOSTNAME Set system host name, default $(b $DEFAULT_HOSTNAME)
   -L LOCALE   Set locale, default: $(b $DEFAULT_LOCALE)
   -P PACKAGES Install extra packages (comma/space separated list)
   -T TOOLSET  The selection of tools to include in the image, default: $(b $(default_toolset))
@@ -273,13 +277,14 @@ Most useful debos options:
 Refer to the README.md for examples
 "
 
-while getopts ":a:b:D:f:hkL:m:P:r:s:T:U:v:x:zZ:" opt; do
+while getopts ":a:b:D:f:hH:kL:m:P:r:s:T:U:v:x:zZ:" opt; do
     case $opt in
         (a) ARCH=$OPTARG ;;
         (b) BRANCH=$OPTARG ;;
         (D) DESKTOP=$OPTARG ;;
         (f) FORMAT=$OPTARG ;;
         (h) echo "$USAGE"; exit 0 ;;
+        (H) HOSTNAME=$OPTARG ;;
         (k) KEEP=true ;;
         (L) LOCALE=$OPTARG ;;
         (m) MIRROR=$OPTARG ;;
@@ -308,6 +313,7 @@ if [ "$ROOTFS" ]; then
     [ "$ARCH"     ] && fail_mismatch -a -r
     [ "$BRANCH"   ] && fail_mismatch -b -r
     [ "$DESKTOP"  ] && fail_mismatch -D -r
+    [ "$HOSTNAME" ] && fail_mismatch -H -r
     [ "$LOCALE"   ] && fail_mismatch -L -r
     [ "$MIRROR"   ] && fail_mismatch -m -r
     [ "$TIMEZONE" ] && fail_mismatch -Z -r
@@ -324,6 +330,7 @@ else
     [ "$ARCH"     ] || ARCH=$DEFAULT_ARCH
     [ "$BRANCH"   ] || BRANCH=$DEFAULT_BRANCH
     [ "$DESKTOP"  ] || DESKTOP=$DEFAULT_DESKTOP
+    [ "$HOSTNAME" ] || HOSTNAME=$DEFAULT_HOSTNAME
     [ "$LOCALE"   ] || LOCALE=$DEFAULT_LOCALE
     [ "$MIRROR"   ] || MIRROR=$DEFAULT_MIRROR
     [ "$TIMEZONE" ] || TIMEZONE=$DEFAULT_TIMEZONE
@@ -470,6 +477,7 @@ echo "# Build options:"
 [ "$TOOLSET"  ] && point "tool selection: $(b $TOOLSET)"
 [ "$PACKAGES" ] && point "additional packages: $(b $PACKAGES)"
 [ "$USERNAME" ] && point "username & password: $(b $USERNAME $PASSWORD)"
+[ "$HOSTNAME" ] && point "hostname: $(b $HOSTNAME)"
 [ "$LOCALE"   ] && point "locale: $(b $LOCALE)"
 [ "$TIMEZONE" ] && point "timezone: $(b $TIMEZONE)"
 [ "$KEEP"     ] && point "keep temporary files: $(b $KEEP)"
